src/app/login/login.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import {FormControl, FormGroup, Validators} from "@angular/forms";
+import {LoginService} from "../services/login.service";
+import {Router} from "@angular/router";
+import {User} from "../model/user";
+import {Credentials} from "../model/credentials";
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+
+  loginForm = new FormGroup({
+    username: new FormControl('', Validators.required),
+    password: new FormControl('', Validators.required)
+  });
+
+  user : User | undefined;
+
+  constructor(private loginService : LoginService, private router: Router) { }
+
+  ngOnInit(): void {
+  }
+
+  onSubmit() {
+    if(this.loginForm.value.username && this.loginForm.value.password) {
+      const cred: Credentials = {
+        username: this.loginForm.value.username,
+        password: this.loginForm.value.password
+      };
+      this.loginService.login(cred
```

Comment:
**suggestion:** take care to avoid memory leaks in your code by unsubscribing from observables. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/